### ISSUE TITLE
Doesn't seem to support comment input

### ISSUE THREAD

**Issue by Kequc** [sindresorhus/domify#45]
Running the following code returns a text node.

`domify("<!-- mycomment -->");`

Seems to be because the function bails early when it does not detect a matching tag.

**Comment by stephenmathieson** [comment]
why would you need to do that?

**Comment by Kequc** [comment]
The classic question, huh. It's because my software loops through nodes from external websites, and so comments are in there. I need to either strip them out, or for my htmltodom script to output them as comments.

Domify outputs comments as text nodes, so you can see them. I don't think that's what you want.

**Comment by stephenmathieson** [comment]
hmm yeah, sounds like a bug. can probably resolve with [`document.createComment`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createComment)

**Comment by Kequc** [comment]
To expand on this. The focus of my software is on Nodes, rather than elements. I've found it often matters `<div></div><div></div>` is sometimes much different from `<div></div> <div></div>` for example. The extra text node can make a difference.

This is especially important when dealing with inline elements, of course. I've found also that `<div></div> <!-- comment --> <div></div>` can cause some unexpected behaviour as well. As that is three nodes between the two divs.

If I were to approach this library I would try to make it Node centric.

I think maybe for the most part it already is, but for example one of the lines strips whitespace from the html. Which could be up to two text nodes that are being displaced.

Perhaps the script could somehow cleverly split the html string up into nodes before figuring out what the content of those nodes were.
